realm/src/main/java/io/realm/Realm.java

```diff
@@ -895,6 +904,80 @@ public void writeCopyTo(File destination) throws IOException {
         return result;
     }
 
+    /**
+     * Converts a RealmObject that has been created using it's standard constructors instead of
+     * {@link #createObject(Class)} to a RealmObject that is managed by the Realm.
+     *
+     * To persist any further changes, the returned object must be be used.
+     *
+     * @param object RealmObject to copy to the Realm.
+     * @return The converted RealmObject that has all its properties managed by the Realm.
+     *
+     * @throws io.realm.exceptions.RealmException if object has already been added to the Realm.
+     */
+    public <E extends RealmObject> E copyToRealm(E object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Null objects cannot be copied into Realm.");
+        }
+
+        // Object is already in this Realm
+        if (object.realm == this) {
+            return object;
+        }
+
+        Class<?> generatedClass;
+        Class<?> objectClass;
+        if (object.realm != null) {
+            // This is already a proxy object from another Realm, get superclass instead (invariant as we don't support subclasses)
+            generatedClass = object.getClass();
+            objectClass = object.getClass().getSuperclass();
+        } else {
+            generatedClass = getProxyClass(object.getClass());
+            objectClass = object.getClass();
+        }
+
+        Method method = copyObjectMethods.get(generatedClass);
+        if (method == null) {
+            try {
+                method = generatedClass.getMethod("copyToRealm", new Class[] {Realm.class, objectClass});
+            } catch (NoSuchMethodException e) {
+                throw new RealmException("Could not find the copyToRealm() method in generated proxy class: " + APT_NOT_EXECUTED_MESSAGE, e);
```

Comment:
Can we call `generatedClass.getName()` at this point? This could make debugger easier for the user.